documentation/design/proposals/global-console-phase-2.adoc

```diff
@@ -62,7 +62,26 @@ It will have the following responsibilities:
 
 ==== GraphQL queries
 
-Root queries.
+The GraphQL interface will provide paginated result sets where necessary.  It will following the offset
+pagination pattern https://graphql.org/learn/pagination/ rather than the cursor one.  The cursor one doesn't meet
+the requirements of the UI.
+
+There will be the following root queries:
+
+```
+addressSpacePlans()                                                  # all defined addressSpacePlans
+addressPlans()                                                       # all defined addressPlans
+addressSpaces(first: Int, offset: Int, filter: String, sort: String) # all visible addressspaces across all viewable namesspaces, paginated
```

Comment:
One of the references mention the cursor based pagination approach. AFAICT, the downside of offset based pagination is that it could lead to outdated views if the underlying result set changes. In our case, maybe that is not something we expect to happen, is that why the cursor based pagination is not used?

Reply:
The offset approach is driving by PF4 requirement design.  It allows you to jump a specific page.  If the client knows the page size and the the number of records, it can calculate the number of pages, and compute the first/offset required to jump to any of those pages.